### DISCUSSION TITLE
Poetry update in synth

### DISCUSSION THREAD

**Discussion by mark-w-325** [projen/projen#3565] [Q&A] [answered]
When you run `npx projen`, it runs poetry update. is there a way to prevent this behavior to freeze the versioning?

**Comment by mrgrain** [comment, marked as answer]
Hi, running the `projen` default task will check if your `pyproject.toml` has changed since the last the synth (*), and if that's the case it will run `poetry update` to ensure that the lockfile is up-to-date. If the file has not changed, `projen` will run the safe `poetry check --lock && poetry install` command. **This feature cannot be disabled easily.**

You can see the logic defined here:
https://github.com/projen/projen/blob/8048cd3dbafe49c695ec64a9dc53e21fdf6d45ea/src/python/poetry.ts#L243-L253


(*) Conceptually, the `pyproject.toml` file is loaded from disk at the beginning of synth, then it's new contents are synth'd and both old and new contents are compared. If there's a difference the file is marked as changed. Most importantly, this could mean that your file was changed by something external to projen. In this case projen will still treat the file as changed.

**Reply by mark-w-325** [reply]
Thanks for the quick response! I will take a look. Right now, I have all packages defined with version=* so that is going to make the install command update the package (I believe). I had some issues with defining the versions and did not need to get tied down with resolving that at the moment.
